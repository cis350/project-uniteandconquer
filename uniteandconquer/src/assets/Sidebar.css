.sidebar {
  margin-bottom: 10px;
  text-align: center;
  margin: 20px;
}

i {
  margin-right: 10px;
}

.greeting-box {
  justify-content: center;
  align-items: center;
  border: 16px solid #ffd9a0;
  padding-bottom: 20px;
  padding-top: 10px;
  margin-bottom: 40px;
}

.greeting {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 10px;

  position: absolute;
  width: 303px;
  height: 98px;
  left: 51px;
  top: 50px;

  background: #fae9c7;
  border: 5px solid #ffd9a0;
  box-sizing: border-box;
}

.link {
  text-decoration: none;
  color: black;
}

.profile-box {
  justify-content: center;
  align-items: center;
  border: 16px solid #ffd9a0;
  margin-bottom: 40px;
}

.profile {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 10px;

  position: absolute;
  width: 303px;
  height: 98px;
  left: 51px;
  top: 167px;

  background: #fae9c7;
  border: 16px solid #ffd9a0;
  box-sizing: border-box;
}

.profile-box:hover {
  background-color: #ffd9a0;
}

.text {
  padding-left: 130px;
  padding-right: 130px;
  padding-top: 30px;
  padding-bottom: 30px;
}

.text-ad {
  padding-left: 130px;
  padding-right: 130px;
  padding-top: 170px;
  padding-bottom: 170px;
}

.chat-box {
  justify-content: center;
  align-items: center;
  border: 16px solid #ffd9a0;
  margin-bottom: 40px;
}

.chat {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 10px;

  position: absolute;
  width: 303px;
  height: 98px;
  left: 51px;
  top: 289px;

  background: #fae9c7;
  border: 16px solid #ffd9a0;
  box-sizing: border-box;
}

.chat-box:hover {
  background-color: #ffd9a0;
}


.login_button {

.logout-box {
  justify-content: center;
  align-items: center;
  border: 16px solid #ffd9a0;
  margin-bottom: 40px;


.logout {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 10px;

  position: absolute;
  width: 303px;
  height: 98px;
  left: 51px;
  top: 412px;

  background: #fae9c7;
  border: 16px solid #ffd9a0;
  box-sizing: border-box;
}


.login_button:hover {

.logout-box:hover {
  background-color: #ffd9a0;
}

.ad-box {
  justify-content: center;
  align-items: center;
  border: 16px solid #ffd9a0;
  margin-bottom: 40px;
}

.ad {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 10px;

  position: absolute;
  width: 303px;
  height: 395px;
  left: 51px;
  top: 535px;

  background: #fae9c7;
  border: 16px solid #ffd9a0;
  box-sizing: border-box;
}

.ad-box:hover {
  background-color: #ffd9a0;
}
