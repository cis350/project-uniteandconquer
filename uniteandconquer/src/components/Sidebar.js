import React from 'react';
import { Link } from 'react-router-dom';
import '../assets/Sidebar.css';

function Sidebar() {
  const myStorage = window.sessionStorage;

  return (
    <div className="sidebar">
      <div className="greeting-box">
        <i className="far fa-user-circle fa-2x" />
        {' '}
        Hi,
        { myStorage.getItem('firstName') ? myStorage.getItem('firstName') : 'Guest'}
      </div>

      <div data-testid="profile-header" className="profile-box">
        <Link className="link" to="/user-profile">
          <div className="text">Profile</div>
        </Link>
      </div>
      <div data-testid="logout-header" className="logout-box">
        <Link className="link" to="/login">
          <div className="text">Login</div>
        </Link>
      </div>
      <div data-testid="ads-header" className="ad-box">
        <Link className="link" to="/">
          <div className="text-ad">Ads</div>
        </Link>
      </div>
    </div>
  );
}

export default Sidebar;
