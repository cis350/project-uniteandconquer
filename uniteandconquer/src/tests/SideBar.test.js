/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import {
  render, screen,
} from '@testing-library/react';
import Sidebar from '../components/Sidebar';

// https://stackoverflow.com/questions/66284286/react-jest-mock-usenavigate

const mockedUsedLink = jest.fn();
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Link: () => mockedUsedLink,
  useNavigate: () => mockedUsedNavigate,
}));

test('sidebar header (profile) content', () => {
  render(<Sidebar />);
  const headerElement = screen.getByTestId('profile-header');
  expect(headerElement).toBeInTheDocument();
});

test('sidebar header (Chat) content', () => {
  render(<Sidebar />);
  const headerElement = screen.getByTestId('chat-header');
  expect(headerElement).toBeInTheDocument();
});

test('sidebar header (Logout) content', () => {
  render(<Sidebar />);
  const headerElement = screen.getByTestId('logout-header');
  expect(headerElement).toBeInTheDocument();
});

test('sidebar header (Ads) content', () => {
  render(<Sidebar />);
  const headerElement = screen.getByTestId('ads-header');
  expect(headerElement).toBeInTheDocument();
});

test('snapshot sidebar', () => {
  const component = renderer.create(<Sidebar />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
