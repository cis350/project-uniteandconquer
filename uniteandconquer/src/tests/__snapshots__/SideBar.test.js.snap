// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot sidebar 1`] = `
<div
  className="sidebar"
>
  <div
    className="greeting-box"
  >
    <i
      className="far fa-user-circle fa-2x"
    />
     
    Hi,
    Guest
  </div>
  <div
    className="profile-box"
    data-testid="profile-header"
  />
  <div
    className="logout-box"
    data-testid="logout-header"
  />
  <div
    className="ad-box"
    data-testid="ads-header"
  />
</div>
`;
